### DISCUSSION TITLE
APIRouter - HTTP Method: Query

### DISCUSSION THREAD

**Discussion by chelodegli** [fastapi/fastapi#5520] [Questions] [answered]
Hi everyone!

I'm looking for a way to use the HTTP Method "Query" with APIRouter. Couldn't find it in the docs, so I assume it has not been implemented. Am I right? Do you know if there is going to be any implementation of it?

Nevertheless, is there a way to customize which methods I could use in my router? I need to receive a "QUERY" method to avoid using "POST" when I'm requesting with too many parameters and conditions to the API.

Links:
[https://datatracker.ietf.org/doc/draft-ietf-httpbis-safe-method-w-body/](https://github.com/encode/httpx/discussions/url)
[https://horovits.medium.com/http-s-new-method-for-data-apis-http-query-1ff71e6f73f3/](https://github.com/encode/httpx/discussions/url)

Thanks for reading!

**Comment by tiangolo** [comment]
It's probably better to write your question in an issue that has all the template and form to guide you writing an example and asking the question in a way that makes it easier to understand and help.

**Comment by odiseo0** [comment]
HTTP Query isn't implemented in FastAPI

**Comment by ghost** [comment]
I was looking for the same thing about [QUERY-HTTP-Method](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-safe-method-w-body-02). I think you should add a [Feature Request] to the Discussion page

**Comment by Kludex** [comment, marked as answer]
You can do the following:

```python
from fastapi import FastAPI

app = FastAPI()


@app.api_route("/", methods=["QUERY"])
async def homepage():
    ...
```

Use `uvicorn` without `httptools`. `httptools` doesn't support the QUERY method.

**Reply by AlePiccin** [reply]
Now that httptools supports it, we should create an PR to include query method in fastapi as well.
